src/LaunchDarkly.EventSource/EventSourceStreamReader.cs

```diff
@@ -1,49 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace LaunchDarkly.EventSource
-{
-    internal interface IStreamReader : IDisposable
-    {
-        Task<string> ReadLineAsync();
-    }
-
-    internal class EventSourceStreamReader : IStreamReader
-    {
-        private StreamReader _streamReader;
-
-        public EventSourceStreamReader(Stream stream, Encoding encoding)
-        {
-            _streamReader = new StreamReader(stream, encoding);
-        }
-
-        public virtual async Task<string> ReadLineAsync()
-        {
-            await Task.Yield();
```

Comment:
I'm really not sure why we created a whole decorator class just to add this one line before each read. If `Task.Yield()` is helpful, we can simply add it before calling `ReadLineAsync()` [here](https://github.com/launchdarkly/dotnet-eventsource/pull/67/files#diff-0e94183352c89bfec8bf05e34839bedc8a4c7ef1adf4348ae42561f87caca04aR156). But I'm not convinced that it is helpful and so I haven't added it there. I think the intention was to in effect make the stream reading task lower-priority, by having it offer up some extra time to other tasks before each of its read operations, but I'm not sure that that's really how .NET task scheduling works— the scheduler should not allow this task to starve other tasks anyway, there is an implicit yield every time you do async I/O— and in any case unless you're getting a very rapid constant flow of stream data, which is not a typical use case for SSE, these reads are not being done very often.